### PR TITLE
Proposal: Duration and time measurement API

### PR DESCRIPTION
**Pull request by ilya-g**
Discussion goes in [KEEP-190](https://github.com/Kotlin/KEEP/issues/190)

Please don't use this pull request for questions, only for proposals how to improve the text.

### CODE REVIEW

File: proposals/stdlib/durations-and-clocks.md [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+- `isNegative()`, `isFinite()`, `isInfinite()` functions
+
+#### Equality and comparison
+
+The equality of Duration is defined so that two duration values representing time intervals of the same length
+are equal, no matter which numeric types and duration units were used to construct these durations, e.g.:
+
+```kotlin
+1.days == 24.hours
+1.5.minutes = 90.seconds
```
Comment (pavlospt) [inline proposals/stdlib/durations-and-clocks.md]: Needs another = here I think!

Comment (ilya-g) [reply proposals/stdlib/durations-and-clocks.md]: Thanks, good catch. Fixed.

**Comment by elizarov** [comment]
It would seem extremely useful to add `String.toDuration()` extension that can reverse both the results of `Duration.toString()` call and the result of `Duration.toIsoString()`.

**Comment by ilya-g** [comment]
@elizarov Please use #190 for discussion. 
Is it ok that `Duration.toString().toDuration()` would be lossy conversion then?

**Pull request merged**